core/src/test/java/com/google/cloud/sql/core/CloudSqlInstanceTest.java

```diff
@@ -186,6 +190,46 @@ public void testCloudSqlInstanceForcesRefresh() throws Exception {
     assertThat(refreshCount.get()).isEqualTo(2);
   }
 
+  @Test
+  public void testCloudSqlInstanceRetriesOnInitialFailures() throws Exception {
+    InstanceData data =
+        new InstanceData(
+            null, new SslData(null, null, null), Instant.now().plus(1, ChronoUnit.HOURS));
+
+    AtomicInteger refreshCount = new AtomicInteger();
+    final PauseCondition refresh0 = new PauseCondition();
+
+    CloudSqlInstance instance =
+        new CloudSqlInstance(
+            "project:region:instance",
+            (instanceName, accessTokenSupplier, authType, executor, keyPair) -> {
+              int c = refreshCount.get();
+              refreshCount.incrementAndGet();
+              switch (c) {
+                case 0:
+                  refresh0.pause();
```

Comment:
A quick comment on why we're pausing on the first request would be helpful here.

Reply:
I guess this isn't strictly needed. I've updated the test accordingly.